src/plugins/chart_expressions/expression_xy/public/expression_renderers/xy_chart_renderer.tsx

```diff
@@ -76,11 +86,60 @@ const extractCounterEvents = (originatingApp: string, layers: CommonXYLayerConfi
       }
     );
 
+    // Multiple axes configured on the same side of the chart
+    const multiAxisSide = Object.values(
+      (yAxisConfigs ?? []).reduce<Record<string, number>>((acc, item) => {
+        if (item.position) {
+          acc[item.position] = (acc[item.position] ?? 0) + 1;
+        }
+        return acc;
+      }, {})
+    ).find((i) => i > 1);
+
+    // There are multiple "split series" aggs in an xy chart and they are not all terms but other aggs
+    const multiSplitNonTerms = new Set(
+      (dataLayer.splitAccessors ?? [])
+        .map(
+          (splitAccessor) =>
+            getColumnByAccessor(splitAccessor, dataLayer.table.columns)?.meta?.params?.id
+        )
+        .filter(Boolean)
+    );
+
+    const aggregateLayers: string[] = dataLayers
+      .map((l) =>
+        l.accessors.reduce<string[]>((acc, accessor) => {
+          const metricType = getColumnByAccessor(
+            accessor,
+            l.table.columns
+          )?.meta?.sourceParams?.type?.toString();
+
+          if (
+            metricType &&
+            ['avg_bucket', 'min_bucket', 'max_bucket', 'sum_bucket'].includes(metricType)
+          ) {
+            acc.push(metricType);
+          }
+          return acc;
+        }, [])
+      )
+      .flat();
+
     return [
       [
         type,
         dataLayer.isPercentage ? 'percentage' : undefined,
         dataLayer.isStacked ? 'stacked' : undefined,
+        // There's a metric configured for the dot size in an area or line chart
```

Comment:
This is concatenating all the strings - can we have them separate instead?
<img width="679" alt="Screenshot 2022-09-15 at 11 52 05" src="https://user-images.githubusercontent.com/1508364/190374479-a65c66f2-fa93-4b6f-bb87-9cf6558cd13a.png">

Shouldn't be `render_agg_based_area_stacked_metric_dot_size_multi_axis_same_side`,

but 
```
render_agg_based_area_stacked_metric_dot_size_multi_axis_same_side
render_agg_based_metric_dot_size
render_agg_based_multi_axis_same_side
```